### PR TITLE
Date factory test when date object is passed

### PR DESCRIPTION
**Pull request by tsteur**
As suggested in chat made this change to see if anything breaks.

There are some failing tests in CustomEvents and Content tracking but that's because of how the date was falsely used in the test fixture. Happy to update the tests and add a changelog entry if we are keen to merge this. Not sure what else it would maybe break that maybe isn't covered by tests.

### CODE REVIEW

File: tests/PHPUnit/Fixtures/ThreeVisitsWithCustomEvents.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ public function trackVisits()
         $this->trackEventWithoutUrl($vis);
         $this->trackMovieWatchingIncludingInterval($vis);
 
-        $this->dateTime = Date::factory($this->dateTime)->addHour(0.5);
+        $this->dateTime = Date::factory($this->dateTime)->addHour(0.5)->getDatetime();
         $vis2 = self::getTracker($this->idSite, $this->dateTime, $useDefault = true, $uselocal);
```
Comment (tsteur) [inline tests/PHPUnit/Fixtures/ThreeVisitsWithCustomEvents.php:64]: we used to pass a `Date` object here meaning the tracker would do `$date->toString()` which then would only result in `YYYY-MM-DD` and throw away the entire time. So this was before actually not working which is why some tests are failing.

File: plugins/Contents/tests/Fixtures/TwoVisitsWithContents.php [hunk 2]
```diff
@@ -47,7 +47,7 @@ public function trackVisits()
 
         $this->trackContentImpressionsAndInteractions($vis);
 
-        $this->dateTime = Date::factory($this->dateTime)->addHour(0.5);
+        $this->dateTime = Date::factory($this->dateTime)->addHour(0.5)->getDatetime();
```
Comment (tsteur) [inline plugins/Contents/tests/Fixtures/TwoVisitsWithContents.php:50]: we used to pass a `Date` object here meaning the tracker would do `$date->toString()` which then would only result in `YYYY-MM-DD` and throw away the entire time. So this was before actually not working which is why some tests are failing.

**Comment by tsteur** [comment]
Any thoughts @sgiehl @diosmosis ?

**Comment by sgiehl** [comment]
In general I would vote for fixing the tests and merging this change. That is actually the behavior I imho would have expected before 🙈

**Comment by tsteur** [comment]
👍 fixed the tests @sgiehl

**Pull request merged**